### PR TITLE
Add test for GPU-AV shader selection using shader objects

### PR DESCRIPTION
**Pull request by Tony-LunarG**
Also fix a GPU-AV problem with shader objects

### CODE REVIEW

File: tests/framework/gpuav_helper.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (spencer-lunarg) [inline tests/framework/gpuav_helper.h:1]: missing copyright

File: tests/unit/shader_object_positive.cpp [hunk 2]
```diff
@@ -1168,6 +1171,130 @@ TEST_F(PositiveShaderObject, ShadersDescriptorSets) {
...
+    TEST_DESCRIPTION("GPU validation: Validate selection of which shaders get instrumented for GPU-AV");
+
+    SetTargetApiVersion(VK_API_VERSION_1_2);
+    const VkBool32 value = true;
+    const VkLayerSettingEXT setting = {OBJECT_LAYER_NAME, "select_instrumented_shaders", VK_LAYER_SETTING_TYPE_BOOL32_EXT, 1,
+                                       &value};
+    VkLayerSettingsCreateInfoEXT layer_settings_create_info = {VK_STRUCTURE_TYPE_LAYER_SETTINGS_CREATE_INFO_EXT, nullptr, 1,
+                                                               &setting};
+    static std::array gpu_av_enables = {VK_VALIDATION_FEATURE_ENABLE_GPU_ASSISTED_EXT};
+    static std::array gpu_av_disables = {VK_VALIDATION_FEATURE_DISABLE_THREAD_SAFETY_EXT,
```
Comment (spencer-lunarg) [inline tests/unit/shader_object_positive.cpp:1184]: there is a `VkGpuAssistedLayerTest::GetValidationFeatures()` 

might be good to put that in the `gpuav_helper.h` file as well

File: tests/unit/shader_object_positive.cpp [hunk 3]
```diff
@@ -1168,6 +1171,130 @@ TEST_F(PositiveShaderObject, ShadersDescriptorSets) {
...
+                                         VK_VALIDATION_FEATURE_DISABLE_CORE_CHECKS_EXT};
+
+    AddRequiredExtensions(VK_EXT_VALIDATION_FEATURES_EXTENSION_NAME);
+    VkValidationFeaturesEXT validation_features = vku::InitStructHelper();
+    validation_features.enabledValidationFeatureCount = size32(gpu_av_enables);
+    validation_features.disabledValidationFeatureCount = 0;  // size32(gpu_av_disables);
+    validation_features.pEnabledValidationFeatures = gpu_av_enables.data();
+    validation_features.pDisabledValidationFeatures = gpu_av_disables.data();
+    validation_features.pNext = &layer_settings_create_info;
+    AddOptionalExtensions(VK_EXT_SHADER_OBJECT_EXTENSION_NAME);
```
Comment (spencer-lunarg) [inline tests/unit/shader_object_positive.cpp:1194]: Required?

File: tests/framework/gpuav_helper.h (hunk 1 above)
Comment (spencer-lunarg) [inline tests/framework/gpuav_helper.h:1]: nit: `gpu_av_helper.h` (to be consistent with the `gpu_av.cpp` test file name

**Comment by jeremyg-lunarg** [comment]
LGTM once Spencer's comments are addressed

**Review by jeremyg-lunarg — APPROVED** [review]

File: tests/unit/shader_object_positive.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply tests/unit/shader_object_positive.cpp]: Moved

File: tests/unit/shader_object_positive.cpp (hunk 3 above)
Comment (Tony-LunarG) [reply tests/unit/shader_object_positive.cpp]: Yes, fixed.

File: tests/framework/gpuav_helper.h (hunk 1 above)
Comment (Tony-LunarG) [reply tests/framework/gpuav_helper.h]: Renamed

**Pull request merged**